RUFAS/biophysical/animal/animal_module_reporter.py

```diff
@@ -138,90 +139,47 @@ def report_daily_animal_population(cls, herd_statistics: HerdStatistics, simulat
         )
 
     @classmethod
-    def report_milk(cls, pen: Pen, simulation_day: int) -> None:
+    def report_milk(cls, milk_reports: list[MilkProductionStatistics], simulation_day: int) -> None:
         """
         Adds milk information for all cows in pen to output manager.
 
         Parameters
         ----------
-        pen : Pen
-            Individual Pen.
+        milk_reports : list[MilkProductionStatistics]
+            A list of MilkProductionStatistics for each lactating cow in the herd.
         simulation_day : int
             Day of simulation.
 
         """
-        units = {
-            "days_in_milk": MeasurementUnits.DAYS,
-            "estimated_daily_milk_produced": MeasurementUnits.KILOGRAMS_PER_DAY,
-            "milk_protein": MeasurementUnits.KILOGRAMS_PER_DAY,
-            "milk_fat": MeasurementUnits.KILOGRAMS_PER_DAY,
-            "milk_lactose": MeasurementUnits.KILOGRAMS_PER_DAY,
-            "lactating": MeasurementUnits.UNITLESS,
-            "parity": MeasurementUnits.UNITLESS,
-            "cow_id": MeasurementUnits.UNITLESS,
-            "pen_id": MeasurementUnits.UNITLESS,
-            "simulation_day": MeasurementUnits.SIMULATION_DAY,
-        }
-
         info_map = {
             "class": AnimalModuleReporter.__name__,
             "function": AnimalModuleReporter.report_milk.__name__,
-            "data_origin": [("Cow", "milking_update")],
-            "units": units,
+            "data_origin": [("MilkProduction", "perform_daily_milking_update")],
+            "units": MilkProductionStatistics.UNITS,
         }
 
-        for animal in list(pen.animals_in_pen.values()):
-            milk_data_update: dict[str, int | float] = {}
-            milk_data_update["days_in_milk"] = animal.days_in_milk
-            milk_data_update["estimated_daily_milk_produced"] = animal.milk_production.daily_milk_produced
-            milk_data_update["milk_protein"] = animal.milk_production.true_protein_content
-            milk_data_update["milk_fat"] = MilkProduction.fat_percent
-            milk_data_update["milk_lactose"] = animal.milk_production.lactose_content
-            milk_data_update["lactating"] = animal.is_milking
-            milk_data_update["parity"] = animal.reproduction.calves
-            milk_data_update["cow_id"] = animal.id
-            milk_data_update["pen_id"] = animal.pen_history[-1]["pen"]
+        for milk_stats in milk_reports:
+            milk_data_update: dict[str, int | float] = asdict(milk_stats)
```

Comment:
```suggestion
            updated_milk_data: dict[str, int | float] = asdict(milk_stats)
```
Took me a second look to understand this variable. I think suggested variable name provides more clarity?